R/InclusionRules.R

```diff
@@ -85,68 +94,55 @@ getInclusionStatisticsFromFiles <- function(cohortIds = NULL,
     signif(delta, 3),
     attr(delta, "units")
   ))
-  if (nrow(result) > 0) {
-    return(result)
-  } else {
-    return(NULL)
-  }
+  inclusionRule <- list(inclusionRuleStats = inclusionRuleStats,
+                        cohortInclusion = inclusion,
+                        cohortIncStats = inclusionStats,
+                        cohortIncResult = inclusionResults,
+                        cohortSummaryStats = summaryStats)
+  return(inclusionRule)
 }
 
 processInclusionStats <- function(inclusion,
                                   inclusionResults,
-                                  simplify,
-                                  inclusionStats,
-                                  summaryStats) {
-  if (simplify) {
-    if (nrow(inclusion) == 0 || nrow(inclusionStats) == 0) {
-      return(NULL)
-    }
-
-    result <- inclusion %>%
-      dplyr::select(.data$ruleSequence, .data$name) %>%
-      dplyr::distinct() %>%
-      dplyr::inner_join(
-        inclusionStats %>%
-          dplyr::filter(.data$modeId == 0) %>%
-          dplyr::select(
-            .data$ruleSequence,
-            .data$personCount,
-            .data$gainCount,
-            .data$personTotal
-          ),
-        by = "ruleSequence"
-      ) %>%
-      dplyr::mutate(remain = 0)
-
-    inclusionResults <- inclusionResults %>%
-      dplyr::filter(.data$modeId == 0)
-    mask <- 0
-    for (ruleId in 0:(nrow(result) - 1)) {
-      mask <- bitwOr(mask, 2^ruleId)
-      idx <-
-        bitwAnd(inclusionResults$inclusionRuleMask, mask) == mask
-      result$remain[result$ruleSequence == ruleId] <-
-        sum(inclusionResults$personCount[idx])
-    }
-    colnames(result) <- c(
-      "ruleSequenceId",
-      "ruleName",
-      "meetSubjects",
-      "gainSubjects",
-      "totalSubjects",
-      "remainSubjects"
-    )
-  } else {
-    if (nrow(inclusion) == 0) {
-      return(list())
-    }
-    result <- list(
-      inclusion = inclusion,
-      inclusionResults = inclusionResults,
-      inclusionStats = inclusionStats,
-      summaryStats = summaryStats
-    )
+                                  inclusionStats) {
+  if (!hasData(inclusion) || !hasData(inclusionStats)) {
+    return(NULL)
   }
+  
+  result <- inclusion %>%
```

Comment:
Its being run. Because of limited data in eunomia, none of the cohort definition when applied to eunomia is giving counts for inclusion rules.

we will have change the cohort definitions being used in tests